### DISCUSSION TITLE
UMD distribution of supabase-js "Supabase is not defined"

### DISCUSSION THREAD

**Discussion by richcorbs** [supabase/supabase#278] [Questions] [answered]
The UMD distribution of supabase.js stopped working recently with the `Supabase is not defined` error.

Previously `Supabase.createClient(..., ...)` worked. Now it doesn't.

What is the new API for supabase-js?

**Comment by soedirgo** [comment]
Hey @richcorbs, sorry we didn't communicate this, we're now relying on [Skypack](https://www.skypack.dev) or [jspm](https://jspm.org) to use supabase-js without a bundler. You can do something like this:

```html
<!DOCTYPE html>
<html>
  <script type="module">
    import { createClient } from "https://cdn.skypack.dev/@supabase/supabase-js";

    const supabase = createClient(
      "https://yourprojectref.supabase.co",
      "your.api.key"
    );

    (async () => {
      const { error, data } = await supabase.from("your_table").select();
      if (error) {
        document.body.innerHTML = `<pre>${JSON.stringify(
          error,
          null,
          2
        )}</pre>`;
      } else {
        document.body.innerHTML = `<pre>${JSON.stringify(data, null, 2)}</pre>`;
      }
    })();
  </script>
</html>
```

**Reply by richcorbs** [reply]
Thank you for the response!

That gets rid of the "Supabase" not defined error but now "supabase" is undefined when I try to use it in my other non-module javascript.

Previously I could initialize "supabase" and then use it inside of Alpine.js. Now I cannot.

Do you know how to make "supabase" from the module available in my non-module javascript?

**Reply by kiwicopple** [reply]
We moved over to typescript on the libraries on 1.0 - looks like we didnt include a UMD build. I'll add the build for the next release

**Reply by kiwicopple** [reply]
Hey @richcorbs , we just did a [patch release](https://github.com/supabase/supabase-js/releases/tag/v1.0.3) of supabase@1.0.3 with Bobbie's updates here: https://github.com/supabase/supabase-js/pull/72

Let us know if that solves it for you!

**Comment by richcorbs** [comment, marked as answer]
Awesome, thanks! supabase@1.0.3 solves it for me!
